src/Microsoft.Azure.WebJobs.Host/Bindings/Data/StructDataBindingProvider.cs

```diff
@@ -8,8 +8,10 @@
 
 namespace Microsoft.Azure.WebJobs.Host.Bindings.Data
 {
+    /// <summary>
+    /// Handles value types (structs) as well as nullable types.
+    /// </summary>
     internal class StructDataBindingProvider<TBindingData> : IBindingProvider
```

Comment:
The switch where we create this provider type is in `DataBindingProvider.CreateTypedBindingProvider`. If for the target type `Type.IsValueType` is true, we create this provider. The problem is a nullable type IS a value type, but is not a struct.